### ISSUE TITLE
Getting STATUS_ILLEGAL_INSTRUCTION on Windows

### ISSUE THREAD

**Issue by ImmemorConsultrixContrarie** [myrrlyn/bitvec#40]
OS x64 Windows, toolchain stable-x86_64-pc-windows-gnu, rustc 1.40.0 (73528e339 2019-12-16).

A subtle bug appeared during tests of my no_std lib. And since [this test](https://github.com/ImmemorConsultrixContrarie/fuzzmer/blob/6a71886389f4fd64c21f46a407f4f82bf2b87c14/src/skimv2_algo/skim_commons.rs#L60) meets no unsafe except for rust internals and bitvec internals, I suspect there's a problem with bitvec. Maybe it's Rust fault itself, but chances are low.
[Comments in the test](https://github.com/ImmemorConsultrixContrarie/fuzzmer/blob/6a71886389f4fd64c21f46a407f4f82bf2b87c14/src/skimv2_algo/skim_commons.rs#L65) describe the problem and [this is the only file](https://github.com/ImmemorConsultrixContrarie/fuzzmer/blob/6a71886389f4fd64c21f46a407f4f82bf2b87c14/src/skimv2_algo/skim_structs.rs) where bitvec is used. There's only six bitvec operations: call to `BitBox::default()` through `derive`, `BitVec::from(BitBox)`, `BitVec.resize()`, `BitVec.into_boxed_bitslice()`, `BitBox.get(usize)`, and `BitBox.get_mut(usize)`.

Forgot to mention: all tests pass on simple "cargo test" in debug mode, and only this one test crashes on "cargo test --release".

**Comment by ImmemorConsultrixContrarie** [comment]
Changed `get_mut()` into `set()`, now everything is fine.

**Comment by myrrlyn** [comment]
Okay that's *really weird*, since `set` and `*get_mut=` should have identical behavior. It looks like something is hitting a `ud2` that only exists in debug? Thanks for leaving this open; this fault is a bug even if it has an immediate workaround.

**Comment by myrrlyn** [comment]
I cannot yet reproduce the test failure on my machines. I do not currently have access to a Windows machine, and running `cargo test --release` on my macOS and Linux machines, on the commit described, does not fail for me.

Can you try patching your worktree to use my WIP branch? You can do so in your manifest with

```toml
# Cargo.toml

[patch.crates-io.bitvec]
git = "https://github.com/myrrlyn/bitvec"
branch = "hotfix/40"
```

or, if that fails, by applying

```diff
diff --git a/src/slice/api.rs b/src/slice/api.rs
index e19d038..b7de50f 100644
--- a/src/slice/api.rs
+++ b/src/slice/api.rs
@@ -2254,7 +2254,11 @@ where O: 'a + BitOrder, T: 'a + BitStore {
 	) -> Self::Mut {
 		BitMut {
 			data: *slice.get_unchecked(self),
-			slot: slice.get_unchecked_mut(self ..= self),
+			slot: {
+				let mut bitptr = slice.bitptr();
+				bitptr.set_len(1);
+				bitptr.into_bitslice_mut()
+			},
 		}
 	}
```

to the copy of `bitvec` in your dependency tree? This is the worse option to actually do, however.

**Comment by ImmemorConsultrixContrarie** [comment]
> if that fails

Still produce the same error. `cargo clean`, added patch, `cargo test --release`, same error.
Cargo clean again, deleted patch from toml, added 

> ```diff
> slot: {
> +				let mut bitptr = slice.get_unchecked_mut(self ..).bitptr();
> +				bitptr.set_len(1);
> +				bitptr.into_bitslice_mut()
> +			},
> ```

to the bitvec-0.17.1\src\slice\api.rs, still same error.

**Comment by myrrlyn** [comment]
uhhh. okay. and using `set` avoids it? that's *really* strange

**Comment by ImmemorConsultrixContrarie** [comment]
> 
> 
> uhhh. okay. and using `set` avoids it? that's _really_ strange

Yep.

**Comment by myrrlyn** [comment]
Would you mind testing if current master (as of b87a81ee28fb8889a568d41d24a4aa58850d8acc) resolves your problem when using `get_mut` instead of `set`? I have not been able to get a target `x86_64-pc-windows-msvc` operational yet, or I would check locally.

I unfortunately still don't know what's actually causing the behavior, so if your changes to `BitMut` don't affect it, I may have to "resolve" this with only a warning in the documentation. Thanks again for bringing it to my attention, even if I'm unable to get it closed.

**Comment by ImmemorConsultrixContrarie** [comment]
Okay, it's fixed. Hella magic!

**Comment by myrrlyn** [comment]
I *profoundly* wish I understood why this solved it, but I'm glad it did. Thanks for the report and test!

**Comment by ImmemorConsultrixContrarie** [comment]
Okay, just retested, it is *definetly* about some change from .17.1, cuz 17.1 from crates still hangs with this error and master-branch passes tests. Yeah, whatever, this is already closed anyway.
> Hella magic!

**Comment by myrrlyn** [comment]
I've published `0.17.2` to crates.io with the changes we've made today.
